services/cd-service/pkg/repository/db.go

```diff
@@ -230,6 +232,115 @@ func (h *DBHandler) DBWriteAllApplications(ctx context.Context, previousVersion
 	return nil
 }
 
+func (h *DBHandler) writeEvent(ctx context.Context, eventuuid, eventType, sourceCommitHash string, eventJson []byte) error {
+	span, ctx := tracer.StartSpanFromContext(ctx, "writeEvent")
+	defer span.Finish()
+	tx, err := h.DB.BeginTx(ctx, nil)
+	if err != nil {
+		return fmt.Errorf("could not begin transaction. Error: %w", err)
+	}
+	insertQuery := h.AdaptQuery("INSERT INTO events (uuid, timestamp, commitHash, eventType, json)  VALUES (?, ?, ?, ?, ?);")
+
+	logger.FromContext(ctx).Sugar().Warnf("Query: %s", insertQuery)
+
+	rawUUID, err := timeuuid.ParseUUID(eventuuid)
+	if err != nil {
+		return fmt.Errorf("error parsing UUID. Error: %w", err)
+	}
+	span.SetTag("query", insertQuery)
+	_, err = tx.Exec(
+		insertQuery,
+		rawUUID.String(),
+		uuid2.GetTime(&rawUUID).AsTime(),
+		sourceCommitHash,
+		eventType,
+		eventJson)
+
+	if err != nil {
+		return fmt.Errorf("Error inserting information into DB. Error: %w\n", err)
+	}
+	err = tx.Commit()
+	if err != nil {
+		return fmt.Errorf("could not commit transaction. Error: %w", err)
+	}
+	return nil
+}
+
+func (h *DBHandler) DBWriteDeploymentEvent(ctx context.Context, uuid, sourceCommitHash, email string, deployment *event.Deployment) error {
+
+	dataJson, err := json.Marshal(deployment)
+
+	if err != nil {
+		return fmt.Errorf("error converting deployment event data to json. Error: %w\n", err)
+	}
+
+	metadataJson, err := json.Marshal(event.Metadata{
+		AuthorEmail: email,
+		Uuid:        uuid,
+	})
+
+	if err != nil {
+		return fmt.Errorf("error converting deployment event metadata to json. Error: %w\n", err)
+	}
+	jsonToInsert, _ := json.Marshal(event.EventJson{
+		DataJson:     string(dataJson),
+		MetadataJson: string(metadataJson),
+	})
+	return h.writeEvent(ctx, uuid, "deployment", sourceCommitHash, jsonToInsert)
+}
+
+func (h *DBHandler) DBSelectAllEventsForCommit(ctx context.Context, commitHash string) ([]EventRow, error) {
+	span, ctx := tracer.StartSpanFromContext(ctx, "DBSelectAllEvents")
+	defer span.Finish()
+
+	query := h.AdaptQuery("SELECT uuid, timestamp, commitHash, eventType, json FROM events WHERE commitHash = (?) ORDER BY timestamp DESC;")
+	span.SetTag("query", query)
+
+	logger.FromContext(ctx).Sugar().Warnf("Query: %s", query)
+	span.SetTag("query", query)
+
+	rows, err := h.DB.QueryContext(ctx, query, commitHash)
+
+	if err != nil {
+		return nil, fmt.Errorf("Error querying DB. Error: %w\n", err)
+	}
+
+	var result []EventRow
+
+	for rows.Next() {
+		var row = EventRow{
+			Uuid:       "",
+			Timestamp:  time.Now(), //will be overwritten, prevents CI linter from complaining from missing fields
```

Comment:
```suggestion
			Timestamp:  time.Unix(0, 0),
```
(nitpick, I just think this is a bit more simple)